### ISSUE TITLE
Party Can't Start Egg Hunt Quest

### ISSUE THREAD

**Issue by MANDRO1D** [HabitRPG/habitica#3263]
My party is stuck with no way to cancel the Egg Hunt quest. Every user has accepted, and I booted a couple users who did not respond. I believe I also booted the user who initiated the quest to see if that would work. 

Is there any way to manually cancel or initiate the quest from the admin side? We're trapped in limbo right now. 

Here's my Group ID: b6fcf6e6-3f25-4c43-b849-bee4a6d6a012

Any help you can provide is hugely appreciated! Thanks!

**Comment by Alys** [comment]
These Tavern posts from the past couple of days might be the same or a similar issue:

From timaeustantalus (uid 2a6f36e8-1411-459d-9944-07e12c7a9d5a): "Hey guys, recently my party got rid of someone who was inactive and who had started a quest for only themselves and not done anything to complete it, but the quest is still active, nothing's happening, and nobody can get rid of said quest. Is there anything at all we can do?"

From spidey588 (00a016ad-d2b2-4070-8853-431aff2fb3ef): "So someone in my party started the egg quest, and wouldn't end it so the rest of us couldn't do our quest. I booted that person from the party, but the quest is still there even though its only participant is gone. Is there a way to completely remove the quest from our party so we can start it again as a group?" [EDIT: That one is reported in #3257 and it's a different group.]

**Comment by MANDRO1D** [comment]
Thanks for providing some additional background, Alys. I booted my party and reinvited everyone, the quest went away as soon as I left the party even though I didn't initiate the quest. Weird.

**Comment by Alys** [comment]
MANDRO1D, nothing in this comment is important, so you can ignore it if you're busy (as can the admins/developers). This is just an explanation for why the fix worked the way it did, in case you're curious.

What you experienced is expected behaviour, although it would seem weird when you don't know how the back end words. When a quest is started, it's under control of the person who started it, but it's linked to the party that that person was in at the time they started it.

When you and everyone else left the party, the party itself would have continued on with that quest still running, but with no people in the party. You'd no longer be able to see that quest because you were no longer in the quest's party.

The new party you created really is a brand new party, even if it has the same name (you'll notice it has a new group ID). Hence it doesn't have the old quest in it.

**Comment by MANDRO1D** [comment]
That totally makes sense of the situation, thanks Alys. Hopefully this issue doesn't affect too many other users. Most of our party has rejoined and we're pretty much back on track.

**Comment by LB7979** [comment]
Possible similar problem here:

Someone in my Party started an Egg Hunt Quest where they were the only participant.
S/he asked if s/he should stop the Quest, and as far as we know was planning to do that.
But now the person who was on the Quest, is suddenly no longer a member of our Party (not sure if they left themselves or if the leader booted them, but in any case they're not here anymore; we also don't know whether or not they aborted the Quest before leaving/booting).

Now we are "in" a Quest but without any participant. How can we stop this?

I really hope booting & re-inviting everyone is not the only option? (BIG party here, would be lot of work for the leader....)

![screenshot 2014-04-14 at 21 49 46](https://cloud.githubusercontent.com/assets/4456266/2700302/5324f4ce-c411-11e3-905c-3a701eef26c0.png)

**Comment by Alys** [comment]
Ryan/deilann posted a solution [elsewhere](https://github.com/HabitRPG/habitrpg/issues/3257). Paraphrasing:

The party leader should collect the User IDs of all the members of the party (click on each person's name in the party screen to find the User ID and copy it to a text file). Then, boot everyone out of the party (warn them first so they know why!), and finally leave the party yourself. Start a new party (you can use the same name; it will still be a new party) and use the User IDs to invite everyone back. The old quest will stay with the old party, and you'll be able to start a new quest in the new party.

**Comment by deilann** [comment]
Even with a big party, it shouldn't be that tough, as you can literally
just put the UUIDs in a doc and copy/paste them into the invite. Make sure
that everyone is in the new party before you try to start the quest again,
though! If someone is not part of the party, they will not get invited when
the scroll is activated.

**Comment by plum1** [comment]
I am having a lot of trouble trying to start the egg quest! My party was together and started the egg quest without me, and then I joined,but now I can't start the quest. My party leader has tried to invite me multiple times. I joined the party primarily for the quest. Would love to start finding some eggs! Help?

**Comment by Alys** [comment]
plum1: After a quest has started, nobody else can join it. You can join a party while a quest is running, but you won't be able to take part in the quest. The options are for you to wait in the party until they start their next quest, or for the person who started the quest (might be a different person than the party leader) to abort the quest and re-start it. 
If you have any other questions, please ask them in Tavern Talk, not in GitHub. GitHub is for reporting bugs not for seeking general help. If you're ever not sure if something is a bug or not, ask in the Tavern first because there's a lot more people there who can answer questions, and someone will direct you to GitHub if it really is a bug.

**Comment by plum1** [comment]
Sorry! I wasn't sure what to do, I did try tavern talk but had posted here
by the time they got back to me. Too bad I can't do the quest. Maybe
something to consider changing in the future? Thanks, and sorry for my
cluelessness.

**Comment by Alys** [comment]
No worries. :) I'd actually just come back here to edit my response because I realised it sounded much harsher than I meant it to. Sorry about that!
It's always worth just waiting a few hours for a response in the Tavern, because not everyone is available to help all the time.
It's very unlikely that this will be changed in future because it could lead to unfair situations. For example, if people join a quest in its final day, they'll receive the same rewards as the people who had been participating for days or weeks, without having had to work hard at their tasks for all that time.

**Comment by plum1** [comment]
That makes sense. Thanks for your help!

**Comment by MANDRO1D** [comment]
I just had the same error happen again with Vice Part 2. We had one hold-out who didn't accept or decline, and I booted the user. The quest didn't auto-start, everyone is listed as accepted. We're back in Quest limbo. 

It's unclear whether the user who initiated the quest has the ability to force it to start, I'll post again once I find out. 

It would be really nice if a Party founder would have the ability to override quest settings. It would be far easier to manage if I could abort or force-start quests that party members initiate, rather than waiting for them to log in and read our party chat.

**Comment by deilann** [comment]
The user who initiated the quest should have the ability to force start.

I disagree with party leaders being able to force start, as it's not their
scroll that it affects, but the quest initiater's.

**Comment by MANDRO1D** [comment]
OK... but for the sake of everyone in the party it would be beneficial to be able to override. Anyone can start a quest and then bail for a week, leaving the party in stasis. I would think that if a user doesn't want to forfeit a scroll, they wouldn't use it in the first place. If a party leader has the right to boot and invite users, manage the party description, or cancel the party altogether, I think it would only be natural that they would also be able to toggle a quest. Maybe if a quest is aborted by the party leader, it could get dropped back into the user's inventory.

**Comment by deilann** [comment]
In this case, I think the party leader should be able to _abort_ a quest,
but not force start it.

**Comment by MANDRO1D** [comment]
Yes, the ability to abort a quest would be awesome. 

We have a rule in my party that all quests have a 24-hour invite window. After that, it's fair game for the initiator to force-start the quest. So far, it has been a positive incentive to stay engaged and check the party chat often.

**Comment by deilann** [comment]
That's a fantastic party rule! The worry I have with force starting from
party leaders goes something like this, which would not apply with your
party, considering its rule, but could be a problem in other parties:

Person starts a quest, goes to bed. They really want their friend to be
part of the quest, too.

Party leader sees that everyone in the party has accepted except the person
the quest starter really wanted to quest with. Party leader goes ahead and
force starts the quest. Why they do, no big deal.

By the time quest starter sees what's going on, other people have dealt
damage/found items, whatever, feels bad, so they don't want to abort and
get rid of the progress. But their friend never got to join in!

The problem here is that quest scrolls are limited or cost money. I know if
I spent a dollar on something I wanted to share with a friend but then
basically couldn't, I would be peeved.

But aborting? I'm all for that.

**Comment by MANDRO1D** [comment]
That definitely makes sense, I can see where the force-start scenario could cause a stir with some groups. Especially given that gems and scrolls are the scarcest resources in the game. 

Thanks for sharing your insights, it's great to know the level to which the dev team is considering the experience of the users.

**Comment by disasterisk** [comment]
I am also experiencing this bug. Got the quest a week before Easter, managed to get everyone in the party to accept by a week after Easter (at no point were we allowed to start the quest without them). It is almost June now and the quest is just kind of sitting there doing nothing.

Group ID: 5579e85d-4dfe-42d9-ad4f-e6d965e2c231

**Comment by Alys** [comment]
disasterisk, can you please show us a screenshot of your party page showing the quest box (the box at the top of the left-hand column that describes the quest) and the recent messages going back a day or two? If there's anything personal in those messages, you can use an image editor to blot out those particular messages using a drawing tool, or cut them out using a rectangular selector and the "cut" feature.

**Comment by disasterisk** [comment]
![quest](https://cloud.githubusercontent.com/assets/5965060/3140919/545e1e40-e94d-11e3-8238-c1974431a822.png)

We haven't exactly said much. I hope this helps!

**Comment by deilann** [comment]
Who initiated the quest?

**Comment by disasterisk** [comment]
Thesemar (d4c8852c-8ea9-46ae-9aca-26326c4ea051)

**Comment by deilann** [comment]
Tell Thesemar to click the start button.

**Comment by Alys** [comment]
Fixed in https://github.com/HabitRPG/habitrpg/pull/4002
Any quests whose quest owner is no longer in the party can now be cancelled/aborted by other party members. More details at that link.
